src/client_capabilities.test.ts

```diff
@@ -0,0 +1,51 @@
+import { assertObjectMatch } from '@std/assert';
+import { describe, it } from '@std/testing/bdd';
+
+import { prepareGetClientCapabilities } from './client_capabilities.ts';
+
+// @ts-ignore For test purposes
+globalThis.PublicKeyCredential = {};
+
+describe('getClientCapabilities', () => {
+  const ua =
+    'Mozilla/5.0 (iPhone; CPU iPhone OS 17_5 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.5 Mobile/15E148 Safari/604.1';
+
+  // @ts-ignore PublicKeyCredential is here for test purposes and never used.
+  PublicKeyCredential.getClientCapabilities = () => {
+    return Promise.resolve({
+      conditionalCreate: true,
+      conditionalMediation: true,
```

Comment:
`conditionalMediation` should likely be `conditionalGet` [in the context of `getClientCapabilities()`](https://w3c.github.io/webauthn/#enumdef-clientcapability) - is `conditionalMediation` testing some older return value?

Reply:
Yeah, Safari 17.4 - 18.2 returns `conditionalMediation` instead of `conditionalGet`. So what it's trying to do here is to
1. Check if the browser is Safari 17.4 - 18.2
2. Get the actual value by executing `getClientCapabilities` from Safari's API
3. Replace `conditionalMediation` with `conditionalGet` and inherit other values from the `getClientCapabilities` result.